From e8fa8d0ca5349d01cf5505eb0d952ef26c62cc94 Mon Sep 17 00:00:00 2001
From: Stephen Webb <swebb@blackberry.com>
Date: Fri, 21 Oct 2022 15:52:22 -0400
Subject: [PATCH] check-namespace.sh: adjust aarch64 symbols

Some symbols for aarch64 were missing, and some were marked as
extraneous in this ABI checker.

Fixes #389.
---
 tests/check-namespace.sh.in | 46 ++++++++++++++++---------------------
 1 file changed, 20 insertions(+), 26 deletions(-)

diff --git a/tests/check-namespace.sh.in b/tests/check-namespace.sh.in
index f99fb59..f9201a2 100644
--- a/tests/check-namespace.sh.in
+++ b/tests/check-namespace.sh.in
@@ -117,7 +117,8 @@ check_local_unw_abi () {
 
     match _U${plat}_flush_cache
     match _U${plat}_get_accessors
-    match _U${plat}_getcontext
+    match _U${plat}_get_elf_image
+    match _U${plat}_get_exe_image_path
     match _U${plat}_regname
     match _U${plat}_strerror
 
@@ -130,68 +131,59 @@ check_local_unw_abi () {
 
     case ${plat} in
 	arm)
-	    match _U${plat}_get_elf_image
-	    match _U${plat}_get_exe_image_path
+	    match _U${plat}_getcontext
 	    match _U${plat}_is_fpreg
 	    match _UL${plat}_search_unwind_table
 	    match _UL${plat}_dwarf_search_unwind_table
 	    match _UL${plat}_dwarf_find_unwind_table
 	    ;;
 	hppa)
+	    match _U${plat}_getcontext
 	    match _UL${plat}_dwarf_search_unwind_table
 	    match _UL${plat}_dwarf_find_unwind_table
-	    match _U${plat}_get_elf_image
-	    match _U${plat}_get_exe_image_path
 	    match _U${plat}_setcontext
 	    ;;
 	ia64)
+	    match _U${plat}_getcontext
 	    match _UL${plat}_search_unwind_table
-	    match _U${plat}_get_elf_image
-	    match _U${plat}_get_exe_image_path
 	    ;;
 	x86)
-	    match _U${plat}_get_elf_image
-	    match _U${plat}_get_exe_image_path
+	    match _U${plat}_getcontext
 	    match _U${plat}_is_fpreg
 	    match _UL${plat}_dwarf_search_unwind_table
 	    match _UL${plat}_dwarf_find_unwind_table
 	    ;;
 	x86_64)
-	    match _U${plat}_get_elf_image
-	    match _U${plat}_get_exe_image_path
+	    match _U${plat}_getcontext
 	    match _U${plat}_is_fpreg
 	    match _UL${plat}_dwarf_search_unwind_table
 	    match _UL${plat}_dwarf_find_unwind_table
 	    match _U${plat}_setcontext
 	    ;;
 	ppc*)
+	    match _U${plat}_getcontext
 	    match _U${plat}_get_func_addr
-	    match _U${plat}_get_elf_image
-	    match _U${plat}_get_exe_image_path
 	    match _U${plat}_is_fpreg
 	    match _UL${plat}_dwarf_search_unwind_table
 	    match _UL${plat}_dwarf_find_unwind_table
 	    ;;
         tilegx)
+	    match _U${plat}_getcontext
             match _U${plat}_is_fpreg
 	    match _UL${plat}_dwarf_search_unwind_table
 	    match _UL${plat}_dwarf_find_unwind_table
             match _UL${plat}_local_addr_space_init
-            match _U${plat}_get_elf_image
-	    match _U${plat}_get_exe_image_path
             match ${plat}_lock
             ;;
 	s390x)
-	    match _U${plat}_get_elf_image
-	    match _U${plat}_get_exe_image_path
+	    match _U${plat}_getcontext
 	    match _U${plat}_is_fpreg
 	    match _UL${plat}_dwarf_search_unwind_table
 	    match _UL${plat}_dwarf_find_unwind_table
 	    match _U${plat}_setcontext
 	    ;;
 	riscv)
-	    match _U${plat}_get_elf_image
-	    match _U${plat}_get_exe_image_path
+	    match _U${plat}_getcontext
 	    match _U${plat}_is_fpreg
 	    match _UL${plat}_dwarf_search_unwind_table
 	    match _UL${plat}_dwarf_find_unwind_table
@@ -288,15 +280,15 @@ check_generic_unw_abi () {
 	    match _U${plat}_dwarf_search_unwind_table
 	    match _U${plat}_dwarf_find_unwind_table
 	    ;;
-        tilegx)
-            match _U${plat}_dwarf_search_unwind_table
+	tilegx)
+	    match _U${plat}_dwarf_search_unwind_table
 	    match _U${plat}_dwarf_find_unwind_table
-            match _U${plat}_get_elf_image
+	    match _U${plat}_get_elf_image
 	    match _U${plat}_get_exe_image_path
-            match _U${plat}_is_fpreg
-            match _U${plat}_local_addr_space_init
-            match ${plat}_lock
-            ;;
+	    match _U${plat}_is_fpreg
+	    match _U${plat}_local_addr_space_init
+	    match ${plat}_lock
+	    ;;
 	s390x)
 	    match _U${plat}_is_fpreg
 	    match _U${plat}_get_elf_image
@@ -315,6 +307,8 @@ check_generic_unw_abi () {
 	    match _U${plat}_is_fpreg
 	    match _U${plat}_dwarf_search_unwind_table
 	    match _U${plat}_dwarf_find_unwind_table
+	    match _U${plat}_get_elf_image
+	    match _U${plat}_get_exe_image_path
 	    ;;
     esac
 
-- 
2.27.0

